rclrs/src/node.rs

```diff
@@ -71,6 +71,9 @@ pub struct Node {
     pub(crate) guard_conditions_mtx: Mutex<Vec<Weak<GuardCondition>>>,
     pub(crate) services_mtx: Mutex<Vec<Weak<dyn ServiceBase>>>,
     pub(crate) subscriptions_mtx: Mutex<Vec<Weak<dyn SubscriptionBase>>>,
+    _clock: Arc<Mutex<Clock>>,
+    // TODO(luca) set to private
+    pub _time_source: Arc<Mutex<Option<TimeSource>>>,
```

Comment:
I believe this is a symptom of one of the pain points I mentioned about how can we create a `TimeSource` in the node builder if it needs to have an `Arc<Node>` as an input parameter. If we solved the issue there this could be made private and the TODO fully removed